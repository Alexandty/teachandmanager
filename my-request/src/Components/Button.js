import React from 'react';

const Button = (props) => (
  <button
    type="button"
    {...props}
    className={`btn btn-dju b-white btn-rounded ${props.className}`}
  >
    {props.children}
  </button>
);

export default Button;