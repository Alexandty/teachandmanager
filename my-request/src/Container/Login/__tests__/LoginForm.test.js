import React from 'react';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Field } from 'redux-form';
import toJson from 'enzyme-to-json';

import { LoginForm } from '../index';
import Button from '../../../Components/Button';

configure({ adapter: new Adapter() });

describe("Probando componente LoginForm", () => {
    it('Probar la accion ', () => {
        const spyLogin = jest.fn();
        const props = {
            login: spyLogin, //validar el login
            handleSubmit: (login) => {
                //fake de la respuesta
                login({ username: 'raul', password: 'alzate' });
            },
            pristine: true
        };
        const wrap = shallow(<LoginForm {...props} />);
        wrap.find(Button).simulate('click')

        expect(spyLogin).toBeCalledWith({ username: 'raul', password: 'alzate' });
    });

    it('Probar los campos ', () => {
        const props = {
            login: jest.fn(),
            handleSubmit: jest.fn(),
            pristine: true
        };
        const wrap = shallow(<LoginForm {...props} />);
        expect(wrap.find(Field).filter({ name: "username" }).exists()).toBe(true);
        expect(wrap.find(Field).filter({
            name: "password",
            type: 'password',
            label: 'Password',
        }).exists()).toBe(true);

        expect(wrap.find(Field).length).toBe(2);
        expect(wrap.find(Button).props()).toEqual({
            type: 'submit',
            disabled: true,
            "className": "my-button",
            children: "LOGIN"
        })
    });

    it('Probar el boton submit ', () => {
        const props = {
            login: jest.fn(),
            handleSubmit: jest.fn(),
            pristine: true
        };
        const wrap = shallow(<LoginForm {...props} />);
        expect(wrap.find(Button).props()).toEqual({
            type: 'submit',
            disabled: true,
            "className": "my-button",
            children: "LOGIN"
        })
    });

    it('Validando toMatchSnapshot', () => {
        const props = {
            login: jest.fn(),
            handleSubmit: jest.fn(),
            pristine: true
        };
        const wrap = shallow(<LoginForm {...props} />);
        expect(toJson(wrap)).toMatchSnapshot();
    });
})