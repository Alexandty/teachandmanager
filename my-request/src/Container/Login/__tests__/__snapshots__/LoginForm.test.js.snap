// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente LoginForm Validando toMatchSnapshot 1`] = `
<Row
  bsClass="row"
  className="show-grid"
  componentClass="div"
>
  <Col
    bsClass="col"
    componentClass="div"
    md={4}
    xs={6}
  />
  <Col
    bsClass="col"
    componentClass="div"
    md={4}
    xs={6}
  >
    <Jumbotron
      bsClass="jumbotron"
      className="Jumbotron"
      componentClass="div"
    >
      <h2
        className="login"
      >
        Login
      </h2>
      <form>
        <div>
          <Field
            component={[Function]}
            label="Username"
            maxLength="20"
            name="username"
            type="text"
          />
        </div>
        <div>
          <Field
            component={[Function]}
            label="Password"
            maxLength="20"
            name="password"
            type="password"
          />
        </div>
        <br />
        <div>
          <Label
            bsClass="label"
            bsStyle="danger"
          />
        </div>
        <br />
        <Button
          className="my-button"
          disabled={true}
          type="submit"
        >
          LOGIN
        </Button>
      </form>
    </Jumbotron>
  </Col>
  <Col
    bsClass="col"
    componentClass="div"
    md={4}
    xsHidden={true}
  />
</Row>
`;
